cmd/xl-v1-object.go

```diff
@@ -512,40 +516,96 @@ func (xl xlObjects) PutObject(bucket string, object string, size int64, data io.
 	// Tee reader combines incoming data stream and md5, data read from input stream is written to md5.
 	teeReader := io.TeeReader(limitDataReader, mw)
 
+	// Initialize parts metadata
+	partsMetadata := make([]xlMetaV1, len(xl.storageDisks))
+
 	// Initialize xl meta.
-	xlMeta := newXLMetaV1(object, xl.dataBlocks, xl.parityBlocks)
+	for index := range partsMetadata {
+		partsMetadata[index] = newXLMetaV1(object, xl.dataBlocks, xl.parityBlocks)
```

Comment:
But it is better to allocate this once and use it when we know this metadata is going to be consistent. So to indicate that distribution is only calculated once per upload. Relatively it would be faster and doesn't employ doing crc32 again and again on the same object name.